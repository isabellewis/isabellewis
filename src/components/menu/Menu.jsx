import React, { Component } from 'react';
import styled from 'styled-components';
import { MenuItem } from '../../elements';
import { SocialMedia } from '../';

const MenuList = styled.ul`
	display: inline-block;
	margin: 0 0 0 auto;
  overflow-x: scroll;
	padding: 0;
	vertical-align: middle;
  white-space: nowrap;

	&.active {
		display: block;
		overflow: hidden;

		@media only screen and (min-width: 576px) {
			display: inline-block;
		}
	}

	&.inactive {
		display: none;

		@media only screen and (min-width: 576px) {
			display: inline-block;
		}
	}
`;

const MenuListWrapper = styled.div`
	text-align: right;

	@media only screen and (min-width: 576px) {
		text-align: left;
	}
`;

const MenuToggleIcon = styled.i`
	cursor: pointer;
	display: inline-block;
	vertical-align: middle;

	@media only screen and (min-width: 576px) {
		display: none;
	}
`;

export class Menu extends Component {
	constructor(props) {
		super(props);

		this.state = {
			menuOpen: false
		}

		this.openMenuToggle = this.openMenuToggle.bind(this);
	}

	openMenuToggle() {
		this.setState (
			{
				menuOpen: !this.state.menuOpen
			}
		);
	}

	render() {
		return (
			<MenuListWrapper>
				<MenuToggleIcon className="material-icons close" onClick={this.openMenuToggle}>{this.state.menuOpen ? 'close' : '' }</MenuToggleIcon>
				<MenuList className={this.state.menuOpen ? 'active' : 'inactive'} onClick={this.openMenuToggle}>
					<MenuItem menuItemText="home" menuItemLocation="heroPanel" />
					<MenuItem menuItemText="about" menuItemLocation="aboutPanel" />
					<MenuItem menuItemText="portfolio" menuItemLocation="portfolioPanel" />
					<MenuItem menuItemText="cv" menuItemLocation="cvPanel" />
				</MenuList>
				<MenuToggleIcon className="material-icons menu" onClick={this.openMenuToggle}>{this.state.menuOpen ? '' : 'menu' }</MenuToggleIcon>
				<SocialMedia theme='mainMenu' />
			</MenuListWrapper>
		);
	}
}
