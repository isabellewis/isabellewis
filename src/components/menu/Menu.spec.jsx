import React from 'react';
import { shallow } from 'enzyme';
import { Menu } from './Menu';

describe('Menu', () => {
	let component;

	beforeEach(() => {
		component = shallow(<Menu title='Menu test text' />);
	});

	it('should match the snapshot', () => {
		expect(component).toMatchSnapshot();
	});
});