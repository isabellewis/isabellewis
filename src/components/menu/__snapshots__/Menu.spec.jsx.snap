// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Menu
    title="Menu test text"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <ForwardRef
          className="material-icons close"
          onClick={[Function]}
        >
          
        </ForwardRef>,
        <ForwardRef
          className="inactive"
          onClick={[Function]}
        >
          <MenuItem
            menuItemLocation="heroPanel"
            menuItemText="home"
          />
          <MenuItem
            menuItemLocation="aboutPanel"
            menuItemText="about"
          />
        </ForwardRef>,
        <ForwardRef
          className="material-icons menu"
          onClick={[Function]}
        >
          menu
        </ForwardRef>,
        <SocialMedia
          theme="mainMenu"
        />,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "",
          "className": "material-icons close",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "Menustyle__MenuToggleIcon-guVoPz",
            "isStatic": true,
            "rules": Array [
              "cursor:pointer;display:inline-block;vertical-align:middle;@media (min-width:576px){display:none;}",
            ],
          },
          "displayName": "Menustyle__MenuToggleIcon",
          "render": [Function],
          "styledComponentId": "Menustyle__MenuToggleIcon-guVoPz",
          "target": "i",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <MenuItem
              menuItemLocation="heroPanel"
              menuItemText="home"
            />,
            <MenuItem
              menuItemLocation="aboutPanel"
              menuItemText="about"
            />,
          ],
          "className": "inactive",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "menuItemLocation": "heroPanel",
              "menuItemText": "home",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "menuItemLocation": "aboutPanel",
              "menuItemText": "about",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "Menustyle__MenuList-hTtCcN",
            "isStatic": true,
            "rules": Array [
              "display:inline-block;margin:0 0 0 auto;padding:0;vertical-align:middle;white-space:nowrap;&.active{display:block;overflow:hidden;@media (min-width:576px){display:inline-block;}}&.inactive{display:none;@media (min-width:576px){display:inline-block;}}",
            ],
          },
          "displayName": "Menustyle__MenuList",
          "render": [Function],
          "styledComponentId": "Menustyle__MenuList-hTtCcN",
          "target": "ul",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": "menu",
          "className": "material-icons menu",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": "menu",
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": undefined,
          "componentStyle": ComponentStyle {
            "componentId": "Menustyle__MenuToggleIcon-guVoPz",
            "isStatic": true,
            "rules": Array [
              "cursor:pointer;display:inline-block;vertical-align:middle;@media (min-width:576px){display:none;}",
            ],
          },
          "displayName": "Menustyle__MenuToggleIcon",
          "render": [Function],
          "styledComponentId": "Menustyle__MenuToggleIcon-guVoPz",
          "target": "i",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "theme": "mainMenu",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": undefined,
      "componentStyle": ComponentStyle {
        "componentId": "Menustyle__MenuListWrapper-bZesvT",
        "isStatic": true,
        "rules": Array [
          "text-align:right;@media (min-width:576px){text-align:left;}",
        ],
      },
      "displayName": "Menustyle__MenuListWrapper",
      "render": [Function],
      "styledComponentId": "Menustyle__MenuListWrapper-bZesvT",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <ForwardRef
            className="material-icons close"
            onClick={[Function]}
          >
            
          </ForwardRef>,
          <ForwardRef
            className="inactive"
            onClick={[Function]}
          >
            <MenuItem
              menuItemLocation="heroPanel"
              menuItemText="home"
            />
            <MenuItem
              menuItemLocation="aboutPanel"
              menuItemText="about"
            />
          </ForwardRef>,
          <ForwardRef
            className="material-icons menu"
            onClick={[Function]}
          >
            menu
          </ForwardRef>,
          <SocialMedia
            theme="mainMenu"
          />,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "",
            "className": "material-icons close",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "Menustyle__MenuToggleIcon-guVoPz",
              "isStatic": true,
              "rules": Array [
                "cursor:pointer;display:inline-block;vertical-align:middle;@media (min-width:576px){display:none;}",
              ],
            },
            "displayName": "Menustyle__MenuToggleIcon",
            "render": [Function],
            "styledComponentId": "Menustyle__MenuToggleIcon-guVoPz",
            "target": "i",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <MenuItem
                menuItemLocation="heroPanel"
                menuItemText="home"
              />,
              <MenuItem
                menuItemLocation="aboutPanel"
                menuItemText="about"
              />,
            ],
            "className": "inactive",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "menuItemLocation": "heroPanel",
                "menuItemText": "home",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "menuItemLocation": "aboutPanel",
                "menuItemText": "about",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "Menustyle__MenuList-hTtCcN",
              "isStatic": true,
              "rules": Array [
                "display:inline-block;margin:0 0 0 auto;padding:0;vertical-align:middle;white-space:nowrap;&.active{display:block;overflow:hidden;@media (min-width:576px){display:inline-block;}}&.inactive{display:none;@media (min-width:576px){display:inline-block;}}",
              ],
            },
            "displayName": "Menustyle__MenuList",
            "render": [Function],
            "styledComponentId": "Menustyle__MenuList-hTtCcN",
            "target": "ul",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": "menu",
            "className": "material-icons menu",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "menu",
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": undefined,
            "componentStyle": ComponentStyle {
              "componentId": "Menustyle__MenuToggleIcon-guVoPz",
              "isStatic": true,
              "rules": Array [
                "cursor:pointer;display:inline-block;vertical-align:middle;@media (min-width:576px){display:none;}",
              ],
            },
            "displayName": "Menustyle__MenuToggleIcon",
            "render": [Function],
            "styledComponentId": "Menustyle__MenuToggleIcon-guVoPz",
            "target": "i",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "theme": "mainMenu",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": undefined,
        "componentStyle": ComponentStyle {
          "componentId": "Menustyle__MenuListWrapper-bZesvT",
          "isStatic": true,
          "rules": Array [
            "text-align:right;@media (min-width:576px){text-align:left;}",
          ],
        },
        "displayName": "Menustyle__MenuListWrapper",
        "render": [Function],
        "styledComponentId": "Menustyle__MenuListWrapper-bZesvT",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
